components/api.ts

```diff
@@ -38,17 +38,11 @@ export const UpdatePollRequest = (userID: string, id: string, data: UpdatePollRe
   })
     .then((response) => {
       if (!response.ok) {
-        return response.json().then(
-          (data) =>
-            Promise.reject({
-              data: null,
-              messages: [],
-            })
-          // TODO: It should probably be this but the backend is not returning errors nicely for 404s and etc.
-          // Promise.reject({
-          //   data: data.data,
-          //   messages: data.messages,
-          // })
+        return response.json().then((data) =>
+          Promise.reject({
+            data: data.data,
+            messages: data.messages,
+          })
```

Comment:
We could just reject data, it wouldn't really change anything other than potentially adding other fields that are passed back in the `data` object. I sort of think of it similarly to spreading props to components, it works but its not recommended to be passing down extra things that may or may not be needed (or removed and then breaking someone that was using it in the future)

Technically we do return a `status` field in an error response that is not included here since we just never use it on the front-end (just legacy from the Go implementation i guess)

Reply:
Yeah that makes sense to me. We should remove the `status` field in the backend since we use the response code for that and we don't use it here. I'll keep a note of that for a future refactor.